packages/css/src/transformCss.ts

```diff
@@ -589,6 +615,42 @@ class Stylesheet {
     }
   }
 
+  transformStartingStyle(
+    root: CSSStyleBlock | CSSSelectorBlock,
+    rules: WithQueries<StyleWithSelectors>['@starting-style'],
+    parentConditions: Array<string> = [],
+  ) {
+    if (rules) {
+      // Check if there are any nested at-rule keys inside this block.
+      // The presence of any key starting with '@' indicates nested queries,
+      // which are not allowed for @starting-style.
```

Comment:
Just flagging that this is actually valid CSS. `@media` inside `@starting-style` is valid ([example](https://codepen.io/askoufis/pen/qEELoQO)), but it relies on your browser supporting CSS nesting.

Actually, looking at the CSS output of your implementation, it generates nested CSS (`@starting-style` inside a selector). However, `@starting-style` is baseline, _and_ it's newer than CSS nesting, so all browsers that support `@starting-style` support CSS nesting, so I think this is fine. However, I think for the sake of simplicity, disallowing _further_ nested CSS within `@starting-style` is the right move for now. We can always remove this restriction in the future if necessary.